TransitInstrument.cs

```diff
@@ -0,0 +1,366 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Lokel.Orbit;
+using System.Linq;
+
+public class TransitInstrument : MonoBehaviour

```

Comment:
this class also feels overloaded. it is doing both detection and taking action based on the detection